debounce/index.js

```diff
@@ -1,5 +1,61 @@
-const { createDebounce } = require('effector-debounce');
+const { is, createEffect, forward, createEvent } = require('effector');
+const { readConfig } = require('../library');
 
-module.exports = {
-  debounce: createDebounce,
-};
+function debounce(argument) {
+  const { source, timeout, target, sid, loc, name } = readConfig(argument, [
+    'source',
+    'timeout',
+    'target',
+
+    'loc',
+    'name',
+    'sid',
+  ]);
+
+  if (!is.unit(source)) throw new Error('source must be unit from effector');
+  if (typeof timeout !== 'number' || timeout < 0)
+    throw new Error('timeout must be positive number or zero');
+
+  const actualName = name || source.shortName || 'unknown';
+
+  // eslint-disable-next-line unicorn/consistent-function-scoping
+  let rejectPromise = () => undefined;
+  let timeoutId;
+
+  const tick =
+    target ||
+    createEvent({
+      name: `${actualName}DebounceTick`,
+      loc,
+    });
+
+  const timerFx = createEffect({
+    name: `${actualName}ThrottleTimer`,
+    sid,
+    loc,
+    handler: (parameter) =>
+      new Promise((resolve, reject) => {
+        rejectPromise = reject;
+        timeoutId = setTimeout(resolve, timeout, parameter);
+      }),
+  });
+
+  timerFx.watch(() => {
+    clearTimeout(timeoutId);
+    rejectPromise();
+  });
+
+  forward({
+    from: source,
+    to: timerFx,
+  });
+
+  forward({
+    from: timerFx.done.map(({ result }) => result),
```

Comment:
```suggestion
    from: timerFx.doneData,
```

Reply:
Also, maybe use `sample` instead of `forward`, just like in `throttle`?

Reply:
But why we need sample? When we need just trigger tick after timer

Reply:
1. to avoid mapping and passing payload to setTimeout
2. to be similar with throttle, and maybe use same code in the future (like lodash uses same function for throttle and debounce)